### ISSUE TITLE
Can't set up custom Ubuntu repo for Adoptium Temurin 

### ISSUE THREAD

**Issue by ppanon2022** [uyuni-project/uyuni#7458]
### Problem description

While Ubuntu has good OpenJDK packages, we need the Temurin packages for support compatibility with Atlassian tools servers. Eclipse set up an Ubuntu/Debian repo for the Adoptium Temurin OpenJDK packages. Standard PPA installation is described in
https://adoptium.net/blog/2021/12/eclipse-temurin-linux-installers-available/

The key instruction is to set up a new apt repo with
echo "deb https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" | tee /etc/apt/sources.list.d/adoptium.list

I tried to set up a custom channel but all I get from a reposync is
2023/08/23 18:04:53 -07:00 Command: ['/usr/bin/spacewalk-repo-sync', '--channel', 'ubuntu-2004-amd64-adoptium', '--type', 'deb', '--non-interactive']
2023/08/23 18:04:53 -07:00 Sync of channel started.
2023/08/23 18:04:53 -07:00 Base URL: https://packages.adoptium.net/ui/native/deb/
2023/08/23 18:04:53 -07:00 Suite: dists/focal/main/binary-amd64/
2023/08/23 18:04:53 -07:00 Component: None
2023/08/23 18:04:53 -07:00 Unhandled error occurred: 'main/binary-amd64/Packages.gz'
2023/08/23 18:04:53 -07:00 Unexpected error: <class 'KeyError'>
2023/08/23 18:04:53 -07:00 Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/reposync.py", line 610, in sync
    http_headers=source_url["http_headers"])
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/repo_plugins/deb_src.py", line 339, in __init__
    self.repo.verify()
  File "/usr/lib/python3.6/site-packages/spacewalk/satellite_tools/repo_plugins/deb_src.py", line 183, in verify
    if not repo.DpkgRepo(self.url, self._get_proxies(), self.gpg_verify).verify_packages_index():
  File "/usr/lib/python3.6/site-packages/spacewalk/common/repo.py", line 435, in verify_packages_index
    entry = self.get_release_index().get(name)
  File "/usr/lib/python3.6/site-packages/spacewalk/common/repo.py", line 77, in get
    return self[key]
KeyError: 'main/binary-amd64/Packages.gz'

### Steps to reproduce

1. Set up a custom channel with
GPG key URL: https://packages.adoptium.net/artifactory/api/gpg/key/public
GPG Key ID:  65F8F04B
GPG Key Fingerprint: 3B04 D753 C905 0D9A 5D34 3F39 843C 48A5 65F8 F04B
2. Set up a repository for the channel with:
Repository URL: https://packages.adoptium.net/ui/native/deb/?uyuni_suite=dists/focal/main/binary-amd64/
Repository type: deb
3. run reposync on the channel
...

### Uyuni version

```bash
Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : Uyuni Server Stable
Name           : Uyuni-Server-release
Version        : 2023.04-220400.204.2.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes
Status         : up-to-date
Source package : Uyuni-Server-release-2023.04-220400.204.2.uyuni2.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.
```

### Additional information

The repo has Release/InRelease files in https://packages.adoptium.net/ui/native/deb/dists/focal/
The repo only has Packages* files (no Release files) in https://packages.adoptium.net/ui/native/deb/dists/focal/main/binary-amd64/
That lack of release files appears to be 
The Packages files has Package file references in pool/main/t/... relative to https://packages.adoptium.net/artifactory/deb

I saw the other issue for setting up a Jenkins custom using <path>/?uyuni_suite=  for an offset flat repo, but that doesn't seem to be applicable here.

**Comment by ppanon2022** [comment]
Sorry, @avshiliaev, you're correct. I was swapping back and forth between tabs to C+P that info and must have swapped to the wrong tab for that and not noticed. I've corrected it. I actually got it right in the Additional information section.
I also tried to use variations like https://packages.adoptium.net/ui/native/deb/?uyuni_suite=dists/focal/main/binary-amd64/ but that didn't help.

**Comment by avshiliaev** [comment]
@ppanon2022 thank you for the explanation!
The stack trace is not quite informative and we probably should add some proper handling here. Anyway, I assume if you run the following commands on the server, the signature is verified just fine for the official Ubuntu repository: 

`wget -qO - http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease | gpg --verify --homedir /var/lib/spacewalk/gpgdir -`

and fails to verify for the custom one:

`wget -qO - https://packages.adoptium.net/ui/native/deb/dists/jammy/InRelease | gpg --verify --homedir /var/lib/spacewalk/gpgdir -`

If so, then you need to import the GPG key on the server before syncing the repo. Please read [Creating Custom Channels and Repositories](https://www.uyuni-project.org/uyuni-docs/en/uyuni/administration/custom-channels.html#_creating_custom_channels_and_repositories)

**Comment by ppanon2022** [comment]
I'm afraid that didn't help. After doing the import the key appears to be in the PGP ring

# gpg --list-keys --keyid-format=long --keyring /var/lib/spacewalk/gpgdir/pubring.gpg
/var/lib/spacewalk/gpgdir/pubring.gpg
-------------------------------------
pub   rsa2048/70AF9E8139DB7C82 2013-01-31 [SC] [expires: 2024-09-20]
      FEAB502539D846DB2C0961CA70AF9E8139DB7C82
uid                 [ unknown] SuSE Package Signing Key <build@suse.de>
.
.
.
pub   rsa2048/843C48A565F8F04B 2021-11-16 [SC]
      3B04D753C9050D9A5D343F39843C48A565F8F04B
uid                 [ unknown] Adoptium GPG Key (DEB/RPM Signing Key) <temurin-dev@eclipse.org>
sub   rsa2048/3F4A517504A9CD61 2021-11-16 [E]

but 
 # wget -qO - https://packages.adoptium.net/ui/native/deb/dists/jammy/InRelease | gpg --verify --homedir /var/lib/spacewalk/gpgdir -
gpg: WARNING: unsafe permissions on homedir '/var/lib/spacewalk/gpgdir'
gpg: no valid OpenPGP data found.
gpg: the signature could not be verified.
Please remember that the signature file (.sig or .asc)
should be the first file given on the command line.

because that wget seems to return junk HTML including </div><noscript><strong>We're sorry but jfrog webapp doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id=app>
Now even if I use a browser to get the InRelease file, save it and try to verify it, I get 
# gpg --verify InRelease
gpg: Signature made Fri Sep  1 06:16:28 2023 PDT
gpg:                using RSA key 843C48A565F8F04B
gpg: BAD signature from "Adoptium GPG Key (DEB/RPM Signing Key) <temurin-dev@eclipse.org>" [unknown]

So it looks like there's two problems 1) they've got crappy Javascript in place which is breaking the download and 2) even then the signature is wrong.
So thanks, for helping me diagnose that, and it doesn't look like there's much that can be done about it from your end. I'll see if I can find somebody at Eclipse to exclude that patch from (what I'm guessing may be) their default css junk.

**Comment by ppanon2022** [comment]
As indicated in the Adoptium github issue I created, the issue is that I was using the wrong URL. The most likely reason is that, as I was investigating the repository URL in a browser, their server rewrote the URL to a different path used for a UI with heavy use of Javascript. When I had identified the path I needed and copied it back into the new repository settings it was therefore leasing to the invalid response noted above.

Understanding the above allowed me to prepare a URL that does work for the Temurin repo.
https://packages.adoptium.net/ui/native/deb/?uyuni_suite=dists/focal/main/binary-amd64/
in the case of Ubuntu 20.04, or 
https://packages.adoptium.net/ui/native/deb/?uyuni_suite=dists/jamy/main/binary-amd64/
for Ubuntu 22.04.
